src/backend/utils/cache/inval.c

```diff
@@ -459,6 +460,7 @@ AddRelcacheInvalidationMessage(InvalidationMsgsGroup *group,
 	msg.rc.id = SHAREDINVALRELCACHE_ID;
 	msg.rc.dbId = dbId;
 	msg.rc.relId = relId;
+	msg.rc.local_only = get_ENR_withoid(currentQueryEnv, relId, ENR_TSQL_TEMP);
```

Comment:
get_ENR_withoid isn't exactly a slow function, but perhaps we can avoid the extra performance overhead by only calling it if we know we are in temp table path? we don't need to do all this if BBF isn't enabled, if temp OID isn't enabled, etc.

Reply:
Hm, maybe we might want to consider enhancing the performance of this lookup in the first place. I redid the test where I did not drop the tables until the end, and it increased the execution time from 52 seconds to 3 minutes 26 seconds. So there is definitely a degradation in performance as the number of simultaneous temp tables increases.

Reply:
After consulting with Alex, it seems that it is extremely unlikely for customers to have more than single digit temp tables active at a single time, so the performance scaling is not much of a concern.

Reply:
if the difference in performance is that drastic (3x worse in that case) then maybe we need to see if we can just have a fast path lookup function for this case.